test_conformance/printf/util_printf.cpp

```diff
@@ -1526,6 +1594,13 @@ static void intRefBuilder(printDataGenParameters& params, char* refResult, const
              atoi(params.dataRepresentation));
 }
 
+static void longRefBuilder(printDataGenParameters& params, char* refResult,
+                           const size_t refSize)
+{
+    snprintf(refResult, refSize, params.genericFormats.front().c_str(),
+             atoll(params.dataRepresentation));
```

Comment:
`params.genericFormats.front().c_str()` will be e.g. `%5ld`, expecting an argument of type `long`, but `atoll` returns a value of type `long long`. On 32-bit platforms, I am seeing this result in
```
0)testing printf("%5ld",10000000000L)
verifyOutputBuffer failed with kernel:
__kernel void test1(void)
{
   printf("%5ld\n",10000000000L);
}

 expected: 1410065408
 got:      10000000000
```
The OpenCL implementation is doing the right thing, but the reference result is wrong, the reference result is based on the low 32 bits of 10000000000.

Could you take a look? Should I open a new issue for it?

Reply:
Thanks @hvdijk , I will review that shortly, no need for new issue